browser/brave_tab_helpers.cc

```diff
@@ -124,6 +130,13 @@ void AttachTabHelpers(content::WebContents* web_contents) {
   }
 
   brave_wallet::BraveWalletTabHelper::CreateForWebContents(web_contents);
+
+#if BUILDFLAG(ENABLE_PLAYLIST)
+  if (playlist::PlaylistServiceFactory::IsPlaylistEnabled(
```

Comment:
it's better to have `MaybeCreateForWebContents` and put this inside

Reply:
Humm, that might bring a layering violation.

```
// static
PlaylistTabHelper::MaybeCreateForWebContents() { << PlaylistTabHelper is in component layer
  if (PlaylistServiceFactory::IsPlaylistEnabled()) { << PlaylistServiceFactory is in browser layer.
  }
}
```

Or, do you think it's still better to have the function in anonymous namespace in this file?

Reply:
you can either put tab_helper into `src/brave/playlist` or make `PlaylistServiceFactory::IsPlaylistEnabled` as a `component/playlist/...`-located function.

Reply:
`PlaylistTabHelper` was located in the component layer as it might be used for Android too. And PlaylistServiceFactory seems to be located in browser layer cause @simonhong didn't want to make the component depend on Profile. I think moving PlaylistServiceFactory into the component worth trying.

Reply:
factories should always be located in brave/browser. Profile is `//chrome/browser` concept, not `//components` one.

Reply:
> Profile is //chrome/browser concept, not //components one

Oh, yeah. I was mistaken about it. I tought there's a something like //component/profile